clients/da-vinci-client/src/main/java/com/linkedin/davinci/config/VeniceServerConfig.java

```diff
@@ -632,6 +643,26 @@ public VeniceServerConfig(VeniceProperties serverProperties, Map<String, Map<Str
     ingestionMemoryLimit = extractIngestionMemoryLimit(serverProperties, ingestionMode, forkedProcessJvmArgList);
     LOGGER.info("Ingestion memory limit: {} after subtracting other usages", ingestionMemoryLimit);
     ingestionMlockEnabled = serverProperties.getBoolean(INGESTION_MLOCK_ENABLED, false);
+
+    try {
+      String producerFactoryClassName = serverProperties.getString(
+          PUB_SUB_PRODUCER_ADAPTER_FACTORY_CLASS,
+          "com.linkedin.venice.pubsub.adapter.kafka.producer.ApacheKafkaProducerAdapterFactory");
```

Comment:
Unless you plan to completely exclude the Kafka dependency from the DVC module (is that the case?) then we should refer to class names programmatically rather than hard coding them, e.g. `ApacheKafkaProducerAdapterFactory.class.getName()`

Reply:
Yeah, we should not hardcode them.